src/scene/batching.js

```diff
@@ -912,9 +1102,9 @@ Object.assign(pc, function () {
         if (dynamic) {
             // Patch the material
             material = material.clone();
-            material.chunks.transformVS = this.transformVS;
-            material.chunks.skinTexVS = this.skinTexVS;
-            material.chunks.skinConstVS = this.skinConstVS;
+            material.chunks.transformVS = this.boneLimit + "#define DYNAMICBATCH\n" + (material.chunks.transformVS ? material.chunks.transformVS : pc.shaderChunks.transformVS);
```

Comment:
Keep in mind that shaders with DYNAMICBATCH define implement skinning with just one bone:
https://github.com/playcanvas/engine/blob/master/src/graphics/program-lib/chunks/transform.vert#L17

If you want proper batched skinning, you need to make shaders use SKIN define instead of it.

Reply:
Hmmm... like: 

    material.chunks.transformVS = this.boneLimit + "#define "+(!isSkin ? "DYNAMICBATCH" : "SKIN")+"\n" + (material.chunks.transformVS ? material.chunks.transformVS : pc.shaderChunks.transformVS);

Reply:
I think all shader-based game engines I ever saw or reverse engineered used 4 bones per-vertex, because it's usually enough, and you don't really want to to compile/switch more shaders just because some objects use less. And 4 bone indices fit neatly into 4 bytes, and you can't have less for a vertex attribute. So PlayCanvas too used 4 bones per vert from the beginning. All imported FBX models are set up to use 4.

Reply:
Ok, i did up some changes you can preview the results in:

https://playcanv.as/p/3QlSEgHF/

Aha, now the animations look much more fluid/real now. Previously, there wasn't any blending. Lol, didn't realised.

Reply:
Pretty cool! I hope what happened to character's leg is expected though :D 

![image](https://user-images.githubusercontent.com/7008423/41585991-4dd690f2-73b4-11e8-9bca-b7a096e721b6.png)

Reply:
Ignore that, urm...was trying to test if IK was possible so i can align it with terrain.....